hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -460,6 +462,11 @@ private void setHmsTableParameters(
               String hmsKey = ICEBERG_TO_HMS_TRANSLATION.getOrDefault(key, key);
               parameters.put(hmsKey, value);
             });
+
+    // 'hms_table_owner' is only used for setting the owner of the table during table creation. No
+    // need to add it as a table property to HMS.
+    parameters.remove(TableProperties.HMS_TABLE_OWNER);
```

Comment:
Any reason why you did not put this into the previous foreach?

Reply:
Nothing specific. Moved this to a filter() before the foreach()